### PR TITLE
Add Stdune.Char

### PR DESCRIPTION
**Pull request by snowleopard**
Following [the suggestion](https://github.com/ocaml/dune/pull/3413#discussion_r413354991) by @rgrinberg.

### CODE REVIEW

File: src/stdune/char.mli [hunk 1]
```diff
@@ -0,0 +1,9 @@
+include module type of struct
+  include Stdlib.Char
+end
+
+(** Check if a character belongs to the set [{'0'..'9'}]. *)
+val is_digit : t -> bool
+
+(** Check if a character belongs to the set [{'0'..'9', 'a'..'f'}]. *)
```
Comment (aalekseyev) [inline src/stdune/char.mli:8]: `'A'..'F'` are generally considered hex as well, so this behavior seems surprising to me.
(sorry for this drive-by comment, I offer no opinion on what the actual semantics in shared-cache should be)

Comment (snowleopard) [reply src/stdune/char.mli:8]: Yep, I was actually also uneasy about this. I'm happy to rename this function to `is_lowercase_hex` to make this clear. Our only current use case is checking whether a directory has a two-character hex name that we use in the cache, and we use lowercase characters, so it seems best to filter out strange names with uppercase characters.

Comment (rgrinberg) [reply src/stdune/char.mli:8]: `is_lowercase_hex` sounds good to me.

Comment (snowleopard) [reply src/stdune/char.mli:8]: Done.
